util/parsefiles.py

```
@@ -0,0 +1,280 @@
+
+            # if their intersection is non-empty, add the docker role
+            if common_roles:
+                items.add(play)
+
+                # each aws role that was in common is marked as being covered by a docker play
+                for role in common_roles:
+                    coverage[role] = True
+
+        self.check_coverage(coverage)
+
+        return items
+
+    def filter_docker_plays(self, plays, repo_path):
+        """Filters out docker plays that do not have a Dockerfile."""
+
+        items = set()
+        logger = logging.getLogger(__name__)
+
+        for play in plays:
+            dockerfile = pathlib2.Path(self.repo_path, "docker", "build", play, "Dockerfile")
+
+            if dockerfile.exists():
+                items.add(play)
+            else:
+                logger.warning(" covered playbook '%s' does not have Dockerfile." % play)
+
+        return items
+
+    def check_coverage(self, coverage):
+        """Checks which aws roles are not covered by docker plays."""
+
+        logging.basicConfig(level=logging.WARNING)
+        logger = logging.getLogger(__name__)
+
+        for role in coverage:
+            if not coverage[role]:
+                logger.warning(" role '%s' is not covered." % role)
+
+    def _get_role_name(self, role):
+            """
+            Resolves a role name from either a simple declaration or a dictionary style declaration.
+
+            A simple declaration would look like:
+            - foo
+
+            A dictionary style declaration would look like:
+            - role: rbenv
+              rbenv_user: "{{ forum_user }}"
+              rbenv_dir: "{{ forum_app_dir }}"
+              rbenv_ruby_version: "{{ forum_ruby_version }}"
+
+            :param role:
+            :return:
+            """
+            if isinstance(role, dict):
+                return role['role']
+            elif isinstance(role, basestring):
+                return role
+            return None
```

Should this print a warning message or throw an exception rather than failing silently?